src/Azure.Functions.Cli/Helpers/AzureHelper.cs

```diff
@@ -359,7 +359,13 @@ private static async Task ArmHttpAsync(HttpMethod method, Uri uri, string access
 
         public static async Task PrintFunctionsInfo(Site functionApp, string accessToken, string managementURL, bool showKeys)
         {
-            var functions = await GetFunctions(functionApp, accessToken, managementURL);
+            ArmArrayWrapper<FunctionInfo> functions = null;
+            await RetryHelper.Retry(async () =>
+            {
+                functions = await GetFunctions(functionApp, accessToken, managementURL);
+                if (!functions.value.Any()) throw new Exception($"GetFunctions is not ready for {functionApp}");
```

Comment:
it is possible to publish a function app without any functions. We should only poll if function app is just published and if the content of the function app locally has any functions

Reply:
Hi @pragnagopa  
I made a change. try not to retry the GetFunctions. Sounds good?